File: C#/sang.cs
using System;

class Program
{
    static void Main()
    {
        string[] game = new string[] { "sang", "kaghaz", "gheychi" };
        string player1, player2;
        int Splayer1 = 0, Splayer2 = 0;
        Random rand = new Random();

        Console.ForegroundColor = ConsoleColor.Yellow;
        Console.Title = "sang";
        Console.SetWindowSize(35, 20);
        Console.Clear();

        while (Splayer1 < 3 && Splayer2 < 3)
        {
            Console.WriteLine("\t    Welcome     \n");
            Console.WriteLine("    sang     kaghaz    gheychi");
            Console.WriteLine("==== shoma {0} va computer {1} ast ====", Splayer1, Splayer2);
            Console.WriteLine("-----------------------------------");
            Console.WriteLine("Enter your choice");
            Console.Write("--->"); player1 = Console.ReadLine();
            while (true)
            {
                bool x = false;
                for (int i = 0; i < 3; i++)
                {
                    if (player1 == game[i])
                    { x = true; break; }
                }
                if (x == false)
                {
                    Console.WriteLine("*******");
                    Console.WriteLine("   use: sang -- kaghaz -- gheychi");
                    Console.WriteLine("*******");
                    Console.Write("--->"); player1 = Console.ReadLine();
                }
                else
                    break;
            }

            Console.WriteLine("-----------------------------------");
            Console.WriteLine("shoma-----> {0}", player1);
            player2 = game[rand.Next(0,3)];
            Console.WriteLine("computer--> {0} \n", player2);

            //-----if
            string win_user = "->shoma barande shodid!";
            string win_computer = "->computer barande shod!";

            if (player1 == player2)
                Console.WriteLine("->mosavi");
            else if (player1 == game[0] && player2 == game[1])
            { Splayer2++; Console.WriteLine(win_computer); }

            else if (player1 == game[0] && player2 == game[2])
            { Splayer1++; Console.WriteLine(win_user); }

            else if (player1 == game[1] && player2 == game[0])
            { Splayer1++; Console.WriteLine(win_user); }

            else if (player1 == game[1] && player2 == game[2])
            { Splayer2++; Console.WriteLine(win_computer); }

            else if (player1 == game[2] && player2 == game[0])
            { Splayer2++; Console.WriteLine(win_computer); }

            else if (player1 == game[2] && player2 == game[1])
            { Splayer1++; Console.WriteLine(win_user); }
            else Console.WriteLine("<-- if ERORR --> ");
            //-----end if

            Console.WriteLine("Press any key to continue . . .");
            Console.ReadKey();
            if (Splayer1 < 3 && Splayer2 < 3)
                Console.Clear();
            else
                continue;
        }

        Console.WriteLine("===================================");
        if (Splayer1 < Splayer2)
        {
            Console.ForegroundColor = ConsoleColor.Red;
            Console.WriteLine("computer bazi ra {0} be {1} bord !!!", Splayer2, Splayer1);
        }
        else if(Splayer1 > Splayer2)
        {
            Console.ForegroundColor = ConsoleColor.Green;
            Console.WriteLine("shoma bazi ra {0} be {1} bordid !!!", Splayer1, Splayer2);
        }

        Console.Write("\nPress any key to Exit ...."); Console.ReadKey();
    }
}

File: C++/console/sang.cpp
#include <iostream>//cin , cout , string
using namespace std;
#include <conio.h> //getch()
#include <stdlib.h> //rand()

string game[3]={"sang", "kaghaz", "gheychi"};
string player1 , player2;
int Splayer1, Splayer2; // in the global int variable  defalt value is 0

int main()
{
	system("color e");
	system("title sang");
	system("mode con cols=35 lines=20");
	system("cls");
	while(Splayer1 < 3 && Splayer2 < 3)
	{
		cout << "\t    Welcome     \n" << endl ;
		cout << "    sang     kaghaz    gheychi" << endl;
		cout <<"==== shoma "<<Splayer1<<" va computer "<<Splayer2<<" ast ===="<<endl;
		cout <<"-----------------------------------"<<endl;
		cout <<"Enter your choice"<<endl;
		cout <<"--->";cin>>player1;
		// check input ----------	
		while (1)
		{
			bool x = false;
			for(int i = 0 ; i<3 ; i++)
			{
				if (player1 == game[i])
					{x = true;break;}
			}
			
			if (x == false)
			{
				cout <<"*******"<<endl;
				cout <<"   use: sang -- kaghaz -- gheychi"<<endl;
				cout <<"*******"<<endl;
				cout <<"--->";cin>>player1;	
			}else
				break;
		}
		// end check input ------
		cout<<"-----------------------------------"<<endl;
		cout<<"shoma-----> "<<player1<<endl; 
		player2 = game[rand()%(3-0+1)+0];
		cout<<"computer--> "<<player2<<"\n"<<endl;
		
		//-----if
		string win_user     = "->shoma barande shodid!" ;
		string win_computer = "->computer barande shod!";
			
	    if (player1 == player2)
	        cout<<"->mosavi"<<endl;
	    else if (player1 == game[0] && player2 == game[1])
	    {Splayer2++; cout<<win_computer<<endl;}
	        
	    else if (player1 == game[0] && player2 == game[2])
		{Splayer1++; cout<<win_user<<endl;}
			
	    else if (player1 == game[1] && player2 == game[0])
		{Splayer1++; cout<<win_user<<endl;}
			
	    else if (player1 == game[1] && player2 == game[2])
		{Splayer2++; cout<<win_computer<<endl;}
			
	    else if (player1 == game[2] && player2 == game[0])
		{Splayer2++; cout<<win_computer<<endl;}
			
	    else if (player1 == game[2] && player2 == game[1])
		{Splayer1++; cout<<win_user<<endl;}	
	    else cout<<"<-- if ERORR --> "<<endl;
		//-----end if
		
		system("pause");
		if(Splayer1 < 3 && Splayer2 < 3)
			system("cls");
		else
			continue;
	}
	cout<<"==================================="<<endl;
	if (Splayer1 < Splayer2)
		{cout<<"computer bazi ra "<<Splayer2<<" be "<<Splayer1<<" bord !!!"<<endl;system("color c");}
	else
		{cout<<"shoma bazi ra "<<Splayer1<<" be "<<Splayer2<<" bordid !!!"<<endl; system ("color a");}
	
	cout<<"\nPress any key to Exit ....";getch();
	return 0 ;
}
